### PR TITLE
Some bug fixes on datajoint pipeline put Anne's codes in a separate directory.

### PR DESCRIPTION
**Pull request by shenshan**

### CODE REVIEW

File: ibl_pipeline/ingest/dj_alter_table.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+"""
+The following is provided as a convenience utility to be able to add/drop non primary key attributes
+to an existing DataJoint table. Both function should work as is - you can simply copy and paste them and
+start using it without worrying about importing dependencies.
+
+WARNING! These functions are NOT officially supported by DataJoint and make use of DataJoint internal logic
+that is considered to be NOT part of the public API. These functions may cease to work or worse yet result
+in unexpected results without any prior notice, and therefore should be used with atmost care.
```
Comment (austin-hilberg) [inline ibl_pipeline/ingest/dj_alter_table.py]:
```suggestion
in unexpected results without any prior notice, and therefore should be used with utmost care.
```

File: ibl_pipeline/ingest/dj_alter_table.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
+"""
+The following is provided as a convenience utility to be able to add/drop non primary key attributes
+to an existing DataJoint table. Both function should work as is - you can simply copy and paste them and
```
Comment (austin-hilberg) [inline ibl_pipeline/ingest/dj_alter_table.py]:
```suggestion
to an existing DataJoint table. Both functions should work as is - you can simply copy and paste them and
```

File: ibl_pipeline/ingest/dj_alter_table.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+start using it without worrying about importing dependencies.
```
Comment (austin-hilberg) [inline ibl_pipeline/ingest/dj_alter_table.py]:
```suggestion
start using them without worrying about importing dependencies.
```

File: ibl_pipeline/ingest/acquisition.py [hunk 4]
```diff
@@ -105,6 +105,8 @@ class WaterAdministrationSession(dj.Manual):
     definition = """
     lab_name:               varchar(255)
```
Comment (dimitri-yatsenko) [inline ibl_pipeline/ingest/acquisition.py]: These primary key attributes are too long.  MySQL has limits on primary key size.

File: ibl_pipeline/ingest/dj_alter_table.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+WARNING! These functions are NOT officially supported by DataJoint and make use of DataJoint internal logic
+that is considered to be NOT part of the public API. These functions may cease to work or worse yet result
+in unexpected results without any prior notice, and therefore should be used with atmost care.
+
+We are working to add such "ALTER" method to DataJoint Python officially, and you can track the discussion/progress
```
Comment (austin-hilberg) [inline ibl_pipeline/ingest/dj_alter_table.py]:
```suggestion
We are working to add such an "ALTER" method to DataJoint Python officially, and you can track the discussion/progress
```

File: ibl_pipeline/ingest/action.py [hunk 6]
```diff
@@ -203,6 +203,8 @@ def make(self, key):
         key['uuid'] = key['surgery_uuid']
 
         subject_uuid = grf(key, 'subject')
+        if len(subject.Subject & 'subject_uuid="{}"'.format(subject_uuid)) == 0:
```
Comment (dimitri-yatsenko) [inline ibl_pipeline/ingest/action.py]:
```suggestion
        if {'subject_uuid': subject_uuid} not in subject.Subject():
```

File: ibl_pipeline/ingest/subject.py [hunk 7]
```diff
@@ -357,6 +357,15 @@ class SubjectProject(dj.Manual):
     subject_nickname:       varchar(255)
     project_name:           varchar(255)
     """
+
+@schema
+class SubjectUser(dj.Manual):
+    definition = """
+    lab_name:               varchar(255)
```
Comment (dimitri-yatsenko) [inline ibl_pipeline/ingest/subject.py]: make shorter.

File: ibl_pipeline/ingest/subject.py [hunk 8]
```diff
@@ -357,6 +357,15 @@ class SubjectProject(dj.Manual):
...
+    subject_nickname:       varchar(255)
+    ---
+    responsible_user:       varchar(255)
```
Comment (dimitri-yatsenko) [inline ibl_pipeline/ingest/subject.py]: use a dependency on users?

File: scripts/ingest_alyx_shadow_membership.py [hunk 9]
```diff
@@ -82,8 +87,23 @@
...
+print('Ingesting subject.SubjectUser...')
+keys = (alyxraw.AlyxRaw & subjects).proj(subject_uuid='uuid')
+for key in keys:
+    key['uuid'] = key['subject_uuid']
+    lab_uuid = grf(key, 'lab')
+    key_su = dict()
+    key_su['lab_name'] = (reference.Lab & 'lab_uuid="{}"'.format(lab_uuid)).fetch1('lab_name')
+    key_su['subject_nickname'] = grf(key, 'nickname')
+    user = grf(key, 'responsible_user')
+    key_su['responsible_user'] = (reference.LabMember & 'user_uuid="{}"'.format(user)).fetch1('user_name')
```
Comment (dimitri-yatsenko) [inline scripts/ingest_alyx_shadow_membership.py]:
```suggestion
    key_su['responsible_user'] = (reference.LabMember & {"user_uuid": user})).fetch1('user_name')
```

File: scripts/ingest_alyx_shadow_membership.py [hunk 10]
```diff
@@ -239,10 +272,13 @@ def make(self, key):
...
 
 # action.SurgeryProcedure
+print('Ingesting action.SurgeryProcedure...')
 keys = (alyxraw.AlyxRaw & 'model = "actions.surgery"').proj(surgery_uuid='uuid')
 for key in keys:
     key_s = dict()
     key['uuid'] = key['surgery_uuid']
+    if len(action.Surgery & key)==0:
+        continue
     key_s['lab_name'], key_s['subject_nickname'], key_s['surgery_start_time'] = \
```
Comment (dimitri-yatsenko) [inline scripts/ingest_alyx_shadow_membership.py]:
```suggestion
    key_s= (action.Surgery & key).fetch(as_dict=True)
```

**Pull request merged**